backend/auth/authToken.go

```diff
@@ -0,0 +1,124 @@
+package auth
+
+import (
+	"bytes"
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"time"
+
+	"github.com/jak103/usu-gdsf/config"
+)
+
+type TokenType int8
+
+const (
+	ACCESS_TOKEN TokenType = iota
+	REFRESH_TOKEN
+)
+
+type TokenClaims struct {
+	Type       TokenType `json:typ`
+	Expiration int64     `json:exp`
+	UserId     int64     `json:uid`
```

Comment:
I'm not sure what data type our user ids will be. They may be integers or UUIDs (which are basically just uint128). This data type would be easy to change.